packages/ra-ui-materialui/src/list/datagrid/DatagridRow.tsx

```diff
@@ -139,7 +139,12 @@ const DatagridRow: FC<DatagridRowProps> = React.forwardRef((props, ref) => {
         <RecordContextProvider value={record}>
             <TableRow
                 ref={ref}
-                className={className}
+                className={clsx(className, {
+                    [DatagridClasses.expandable]: expandable,
+                    [DatagridClasses.selectable]: selectable,
+                    [DatagridClasses.clickableRow]:
+                        typeof rowClick === 'function' ? false : rowClick,
```

Comment:
I believe this should rather be

```suggestion
                        typeof rowClick === 'function' ? true : rowClick,
```